fvtest/porttest/omrsockTest.cpp

```diff
@@ -237,7 +237,40 @@ TEST(PortSockTest, create_hints_and_element_extraction)
  */
 TEST(PortSockTest, getaddrinfo_and_freeaddrinfo)
 {
-	/* Unimplemented. */
+	OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+
+	OMRAddrInfoNode result;
+	omrsock_addrinfo_t hints = NULL;
+	int32_t rc = 0;
+	uint32_t length = 0;
+	int32_t hintsFamily = 0;
+	int32_t hintsSockType = 1;
+	int32_t hintsProtocol = 0;
+	int32_t hintsFlags = 0;
+
+	rc = OMRPORTLIB->sock_getaddrinfo_create_hints(OMRPORTLIB, &hints, hintsFamily, hintsSockType, hintsProtocol, hintsFlags);
+	EXPECT_EQ(rc, 0);
+
+	/* Testing invalid arguments: getaddrinfo pointer to result is NULL. */
+	omrsock_addrinfo_t resultPtr = NULL;
+
+	rc = OMRPORTLIB->sock_getaddrinfo(OMRPORTLIB, (char *)"localhost", NULL, hints, resultPtr);
```

Comment:
Is `(char *)` needed?
```suggestion
	rc = OMRPORTLIB->sock_getaddrinfo(OMRPORTLIB, "localhost", NULL, hints, resultPtr);
```

Reply:
@babsingh I remembered why I added it in, it seems like the conversation from string to char* is deprecated for some reason. Should I add it back in?
```/root/omr/fvtest/porttest/omrsockTest.cpp:261:83: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
  rc = OMRPORTLIB->sock_getaddrinfo(OMRPORTLIB, "localhost", NULL, hints, resultPtr);
                                                                                   ^
/root/omr/fvtest/porttest/omrsockTest.cpp:269:81: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
  rc = OMRPORTLIB->sock_getaddrinfo(OMRPORTLIB, "localhost", NULL, hints, &result);
                                                                                 ^
/root/omr/fvtest/porttest/omrsockTest.cpp:291:80: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
  rc = OMRPORTLIB->sock_getaddrinfo(OMRPORTLIB, "localhost", NULL, NULL, &result);
```